File: itexpert-content-ui/src/app/modeles/Node.ts
import {Value} from "./Value";
import {Rule} from "./Rule";
import {ContentNode} from "./ContentNode";
import {Translation} from "./Translation";

export class Node {
  public id: string;
  public parentCode: string;
  public parentCodeOrigin: string;
  public name: string;
  public code: string;
  public environmentCode: string;
  public description: string;
  public defaultLanguage: string;
  public type: string;
  public subNodes: string[];
  public creationDate: number;
  public modificationDate: number;
  public modifiedBy: string;
  public snapshot: boolean;

  public tags: string[];
  public values: Value[];
  public roles: string[];
  public rules: Rule[];
  public contents: ContentNode[];
  public languages: string[];

  public publicationDate: number;
  public version: string;
  public status: string;
  public publicationStatus: string;

  public translations: Translation[];

  public userName: string;
}

File: itexpert-content-lib/src/main/java/com/itexpert/content/lib/entities/Node.java
package com.itexpert.content.lib.entities;

import com.itexpert.content.lib.enums.StatusEnum;
import com.itexpert.content.lib.models.Rule;
import com.itexpert.content.lib.models.Translation;
import com.itexpert.content.lib.models.Value;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Document(collection = "nodes")
@Data
public class Node implements Serializable, Cloneable {

    @Id
    private UUID id;

    private String code;

    private String parentCode;

    private String parentCodeOrigin;

    private String name;

    private String description;

    private String defaultLanguage;

    private String type;

    private List<String> subNodes;


    private List<String> tags;

    private List<Value> values;

    private List<String> roles;

    private List<Rule> rules;

    private List<String> languages;

    private Long creationDate;

    private Long modificationDate;

    private UUID modifiedBy;


    private String version;
    private Long publicationDate;
    private StatusEnum status;

    private boolean favorite;

    private List<Translation> translations;

    public Node clone() throws CloneNotSupportedException {
        return (Node) super.clone();
    }
}
